backend/src/zimfarm_backend/api/routes/dependencies.py

```diff
@@ -64,7 +88,22 @@ def _get_current_user_or_none(
     ) -> User | None:
         if claims is None:
             return None
-        return get_user_by_id_or_none(session, user_id=claims.subject)
+        user = get_user_by_id_or_none(session, user_id=claims.sub)
+        if user is None:
+            user = get_user_by_idp_sub_or_none(session, idp_sub=claims.sub)
```

Comment:
Why not merging `get_user_by_idp_sub_or_none` into `get_user_by_id_or_none` and simply doing an "OR" on `id` and `idp_sub` fields? since both are very unique, I don't see any risk and it keeps the code a bit cleaner